x-pack/plugins/uptime/public/components/common/react_router_helpers/eui_link.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import {
+  EuiLink,
+  EuiButton,
+  EuiButtonProps,
+  EuiButtonEmptyProps,
+  EuiLinkAnchorProps,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+
+import { letBrowserHandleEvent } from './link_events';
+
+/**
+ * Generates either an EuiLink or EuiButton with a React-Router-ified link
+ *
+ * Based off of EUI's recommendations for handling React Router:
+ * https://github.com/elastic/eui/blob/master/wiki/react-router.md#react-router-51
+ */
+
+interface IEuiReactRouterProps {
+  to: string;
+}
+
+export const EuiReactRouterHelper: React.FC<IEuiReactRouterProps> = ({ to, children }) => {
```

Comment:
I'm questioning this naming convention. These components aren't actually part of EUI so I am concerned that overloading their prefix could be a source of confusion.

Perhaps we can refer to them as `Uptime{Helper}` or `Um{Helper}` components.

Reply:
i have removed the Eui prefix, i think it makes more sense now.

Reply:
I might not have explained what I meant clearly enough - I was talking in general, we should probably avoid prefixing anything that we ourselves maintain with the `Eui` prefix because it overloads the meaning.

If someone sees a reference to `EuiReactRouterHelper` in our code they may try to find documentation that doesn't exist. Renaming it as you have below (like `ReactRouterEuiHelper`) might be a more effective solution. Does that make sense? cc @andrewvc

Reply:
ahh i agree, i renamed others but i missed this. let me do that.

Reply:
👍 ping me when it's ready for another look and we are otherwise in pretty good shape here I think